config/yamlflag.go

```diff
@@ -0,0 +1,42 @@
+package config
+
+import (
+	"fmt"
+
+	"gopkg.in/yaml.v2"
+)
+
+type yamlFlag[T any] struct {
+	Ptr   **T
```

Comment:
ouch :)
But I see why

Reply:
The usage looks like
```go
type Config struct {
  OpenTelemetry *otel.Options `yaml:"open-telemetry"`
  ...
}
...
flag.Var(newYamlFlag(&cfg.OpenTelemetry), "open-telemetry", "OpenTelemetry configuration in YAML format, use flow-style for convenience")
...

if o.OpenTelemetry != nil {
...
}
```

The value is allocated when flag is set and stays nil when it is not, i.e. this allows to tell if flag was provided.

Reply:
Yes I saw it, all good  :)